independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeResourceDeployment.java

```diff
@@ -324,6 +326,10 @@ public RuntimeResource buildResourceMethod(ResourceClass clazz,
             }
             handlers.add(new RequestDeserializeHandler(typeClass, genericType, consumesMediaTypes, serialisers,
                     bodyParameterIndex));
+            if (inputHandlerEngaged) {
+                handlers.add(new NonBlockingHandler());
```

Comment:
I wonder if this should be
```suggestion
                handlers.add(NonBlockingHandler.INSTANCE);
```
like in line 232 above?

Reply:
Good point :)